### PR TITLE
Update appointment command for sorting and update Developer Guide to include appointment feature

### PR DESCRIPTION
**Pull request by Dineshraj555**

### CODE REVIEW

File: src/main/java/seedu/address/logic/parser/ParserUtil.java [hunk 1]
```diff
@@ -133,43 +129,21 @@ public static Tag parseTag(String tag) throws ParseException {
...
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("dd-MM-yyyy HH:mm");
         try {
-            DateTimeFormatter formatIn = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm");
-            DateTimeFormatter formatOut = DateTimeFormatter.ofPattern("MMM-dd-yyyy HH:mm a");
-            return LocalDateTime.parse(input, formatIn).format(formatOut);
+            return LocalDateTime.parse(trimmedDateTime, formatter);
         } catch (Exception e) {
-            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT,
-                    AppointmentCommand.MESSAGE_USAGE), e);
+            throw new ParseException("Appointment date and time entered the date-time in dd-MM-yyyy HH:mm format");
```
Comment (tsumian) [inline src/main/java/seedu/address/logic/parser/ParserUtil.java:146]: Can consider changing this message to make it clearer. At first glance was not clear what this does from the error message.

File: src/test/java/seedu/address/testutil/PetBuilder.java [hunk 2]
```diff
@@ -108,10 +109,20 @@ public PetBuilder withDiet(String diet) {
...
+        return this;
+    }
+
+    /**
+     * Sets the {@code Appointment} of the {@code Pet} that we are building with the input.
+     */
+    public PetBuilder withAppointment(String dateTime, String location) {
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("dd-MM-yyyy HH:mm");
+        LocalDateTime formattedDateTime = LocalDateTime.parse(dateTime, formatter);
+        this.appointment = new Appointment(formattedDateTime, location);
```
Comment (tsumian) [inline src/test/java/seedu/address/testutil/PetBuilder.java:125]: Cleaner alternative than the previous implementation!

**Review by tsumian — COMMENTED** [review]
Just one slight modification to be considered for the code to be more comprehensive. Otherwise LGTM!

File: src/main/java/seedu/address/logic/parser/ParserUtil.java (hunk 1 above)
Comment (Dineshraj555) [reply src/main/java/seedu/address/logic/parser/ParserUtil.java:146]: Agreed

File: src/main/java/seedu/address/model/pet/Appointment.java [hunk 3]
```diff
@@ -1,40 +1,122 @@
...
+        this.value = "";
+        this.dateTime = null;
+        this.location = null;
+    }
+
+    public LocalDateTime getDateTime() {
+        return this.dateTime;
+    }
+
+    public String getLocation() {
```
Comment (tobihy) [inline src/main/java/seedu/address/model/pet/Appointment.java:76]: Just realised, but there's some missing Javadoc comments here!

**Pull request merged**